var/spack/repos/builtin/packages/vtk/package.py

```diff
@@ -59,6 +62,9 @@ class Vtk(CMakePackage):
     # VTK 8.1, that should change
     conflicts('+osmesa', when='+qt')
 
+    # mpi4py calls seem not to work with py37
+    conflicts('+python', when='^python@3.7')
```

Comment:
this should probably go into the `py-mpi4py` package.
also, it's missing in the current `depends_on` dependencies, should it be added?

Reply:
Ok, should I move it to py-mpi4py so? I'm no sure it's a problem of `mpi4py` or vtk being not up to date...

Reply:
Since the `vtk` package seems to *use* `mpi4py` but does not declare it as a dependency yet, I guess it is a `mpi4py` package from your system that is used here.

Can you try to add:
```
depends_on('py-mpi4py', when='+mpi +python')
```
and see if this already solves the problem you saw with Python 3.7?

Reply:
Still getting lots of this: 
```
In file included from /md1200-1/home/CMAP/ruben.di-battista/.local/spack/var/spack/stage/vtk-8.1.1-iyozqpypng4w2fhavm5hock3bz5ztarc/VTK-8.1.1/ThirdParty/mpi4py/vtkmpi4py/src/MPI.c:4:
     699     /md1200-1/home/CMAP/ruben.di-battista/.local/spack/var/spack/stage/vtk-8.1.1-iyozqpypng4w2fhavm5hock3bz5ztarc/VTK-8.1.1/ThirdParty/mpi4py/vtkmpi4py/src/mpi4py.MPI.c: In function ‘__Pyx_PyCFu
             nction_FastCall’:
  >> 700     /md1200-1/home/CMAP/ruben.di-battista/.local/spack/var/spack/stage/vtk-8.1.1-iyozqpypng4w2fhavm5hock3bz5ztarc/VTK-8.1.1/ThirdParty/mpi4py/vtkmpi4py/src/mpi4py.MPI.c:163017:13: error: too man
             y arguments to function ‘(PyObject * (*)(PyObject *, PyObject * const*, Py_ssize_t))meth’
```

Reply:
I actually needed to force `VTK_USE_SYSTEM_MPI4PY=ON`. But now I'm getting other errors:
```
  >> 2249    .local/spack/var/spack/stage/vtk-8.1.1-iyozqpypng4w2fhavm5hock3bz5ztarc/VTK-8.1.1/Wrapping/PythonCore/vtkPythonArgs.cxx:105:25: error: invalid conversio
             n from ‘const char*’ to ‘char*’ [-fpermissive]

```

And it looks a common thing with Python 3.7: https://github.com/tensorflow/tensorflow/issues/20517

Reply:
I backported the fix from upstream VTK for `python3.7`. On my machine now compiles correctly.